tests/e2e/vsphere.go

```diff
@@ -89,6 +90,59 @@ func (vs *vSphere) queryCNSVolumeWithResult(fcdID string) (*cnstypes.CnsQueryRes
 	return &res.Returnval, nil
 }
 
+// queryCNSVolumeSnapshotWithResult Call CnsQuerySnapshots
+// and returns CnsSnapshotQueryResult to client
+func (vs *vSphere) queryCNSVolumeSnapshotWithResult(fcdID string,
+	snapshotId string) (*cnstypes.CnsSnapshotQueryResult, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// Connect to VC
+	connect(ctx, vs)
+	var snapshotSpec []cnstypes.CnsSnapshotQuerySpec
+	snapshotSpec = append(snapshotSpec, cnstypes.CnsSnapshotQuerySpec{
+		VolumeId: cnstypes.CnsVolumeId{
```

Comment:
are both volumeid and snapshotid mandatory?
can two volumeids share a snapshot?

Reply:
Yes they are. passing an empty or no snapshot id returns no snapshots
Two volumes cannot share the same snapshot - snapshot and volumes has 1 to 1 mapping.

Passing either of them as empty or not passing at all returns error

13:52:56    Unexpected error:
13:52:56        <soap.soapFaultError>: {
13:52:56            fault: {
13:52:56                XMLName: {
13:52:56                    Space: "http://schemas.xmlsoap.org/soap/envelope/",
13:52:56                    Local: "Fault",
13:52:56                },
13:52:56                Code: "ServerFaultCode",
13:52:56                String: "snapshotQuerySpec.volumeId.Id is empty.",
13:52:56                Detail: {
13:52:56                    Fault: <types.InvalidArgument>{
13:52:56                        RuntimeFault: {
13:52:56                            MethodFault: {FaultCause: nil, FaultMessage: nil},
13:52:56                        },
13:52:56                        InvalidProperty: "snapshotQuerySpec.volumeId.Id",
13:52:56                    },
13:52:56                },
13:52:56            },
13:52:56        }
13:52:56        ServerFaultCode: snapshotQuerySpec.volumeId.Id is empty.
13:52:56    occurred
13:52:56